dGame/dComponents/PetComponent.h

```diff
@@ -349,7 +349,7 @@ class PetComponent final : public Component
 	/**
 	 * Preconditions that need to be met before an entity can tame this pet
 	 */
-	PreconditionExpression* m_Preconditions;
+	std::optional<PreconditionExpression> m_Preconditions = std::nullopt;
```

Comment:
```suggestion
	std::optional<PreconditionExpression> m_Preconditions;
```
This should achieve the same result?

Reply:
If I'm going to default-initialize I think I should actually go with
```suggestion
	std::optional<PreconditionExpression> m_Preconditions{};
```